services/CurrentTime/include/ble-service-current-time/CurrentTimeService.h

```diff
@@ -0,0 +1,178 @@
+/* mbed Microcontroller Library
+ * Copyright (c) 2020 ARM Limited
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef BLE_CURRENT_TIME_SERVICE_H
+#define BLE_CURRENT_TIME_SERVICE_H
+
+#ifdef BLE_FEATURE_GATT_SERVER
+
+#include "ble/BLE.h"
+#include "ble/Gap.h"
+#include "events/EventQueue.h"
+#include "mbed_rtc_time.h"
+
+#include <time.h>
+
+/**
+ * BLE Current Time Service
+ *
+ * @par purpose
+ *
+ * @par usage
+ *
+ * @note The specification for the current time service can be found here:
+ * https://www.bluetooth.com/specifications/gatt
+ *
+ * @attention The user should not instantiate more than a single current time service service
+ */
+class CurrentTimeService : private ble::Gap::EventHandler {
+public:
+    struct CurrentTime {
+        uint16_t year;
+        uint8_t  month;
+        uint8_t  day;
+        uint8_t  hours;
+        uint8_t  minutes;
+        uint8_t  seconds;
+
+        uint8_t  weekday;
+
+        uint8_t  fractions256;
+
+        uint8_t  adjustReason;
```

Comment:
some comments about the values would help understand the contents next time someone has to edit this
```suggestion
        uint16_t year; // calendar year based at year 1, with valid values 1582-9999
        uint8_t  month; // 1-12
        uint8_t  day; //calendar day
        uint8_t  hours;
        uint8_t  minutes;
        uint8_t  seconds;

        uint8_t  weekday; // 1-7 starting with Monday

        uint8_t  fractions256;

        uint8_t  adjustReason;
```